recipe/meta.yaml

```diff
@@ -13,43 +14,235 @@ source:
 build:
   number: 0
   noarch: python
-  script: {{ PYTHON }} -m pip install . --no-deps -vv
-  entry_points:
-    - parsl-globus-auth = parsl.data_provider.globus:cli_run
-
-requirements:
-  host:
-    - python >=3.5
-    - pip
-  run:
-    - python >=3.5
-    - pyzmq >=17.1.2
-    - typeguard >=2.5
-    - ipyparallel
-    - globus-sdk
-    - dill
-    - tblib
-    - ipykernel
-    - requests
-    - paramiko
-    - psutil >=5.5.1
-    - sqlalchemy >=1.3.0,!=1.3.4,<1.4
-    - sqlalchemy-utils
-    - pydot
-    - networkx
-    - flask >=1.0.2
-    - flask-sqlalchemy
-    - pandas
-    - plotly
-    - python-daemon
-
-test:
-  requires:
-    - pip
-  imports:
-    - parsl
-  commands:
-    - pip check
+
+outputs:
+  - name: {{ name }}
+    script: {{ PYTHON }} -m pip install . --no-deps -vv
+    build:
+      noarch: python
+      entry_points:
+        - parsl-globus-auth = parsl.data_provider.globus:cli_run
+
+    requirements:
+      host:
+        - python >=3.5
+        - pip
+      run:
+        - python >=3.5
+        # requirements.txt
+        - pyzmq >=17.1.2
+        - typeguard >=2.10
+        - typing-extensions
+        - globus-sdk
+        - dill
+        - tblib
+        - requests
+        - paramiko
+        - psutil >=5.5.1
+
+    test:
+      requires:
+        - pip
+      imports:
+        - parsl
+      commands:
+        - pip check
+
+  - name: {{ name }}-with-monitoring
+    requirements:
+      host:
+        - python
+      run:
+        - python
+        - {{ pin_subpackage(name, max_pin='x.x.x.x.x.x') }}
+        - sqlalchemy >=1.3.0,!=1.3.4,<1.4
+        - sqlalchemy-utils
+        - pydot
+        - networkx
+        - flask >=1.0.2
+        - flask-sqlalchemy
+        - pandas
+        - plotly
+        - python-daemon
+    test:
+      imports:
+        - parsl
+      commands:
+        - pip check
```

Comment:
@yadudoc and @ZhuozhaoLi, for each of the new packages, do you have a good import (or command-line) test we can run to make sure the extra is working as expected?  Just importing `parsl` itself in each test is not very rigorous, so it would be nice to do better if it's not too hard.